crates/sui-core/src/consensus_adapter.rs

```diff
@@ -170,18 +180,127 @@ impl SubmitToConsensus for TransactionsClient<sui_network::tonic::transport::Cha
     }
 }
 
+pub struct ConnectionMonitorStatusForTests {}
+
+impl CheckConnection for ConnectionMonitorStatusForTests {
+    fn check_connection(&self, _authority: &AuthorityName) -> Option<ConnectionStatus> {
+        Some(ConnectionStatus::Connected)
+    }
+}
+
+pub trait CheckConnection {
+    fn check_connection(&self, authority: &AuthorityName) -> Option<ConnectionStatus>;
+}
+
+pub struct ConnectionMonitorStatus {
+    /// Current connection statuses forwarded from the connection monitor
+    pub connection_statuses: Arc<DashMap<AuthorityName, ConnectionStatus>>,
```

Comment:
Similar to my other comment bellow about the epoch change with committee change - unless we create a new `ConnectionMonitorStatus` per epoch, this map will never get cleared up with members that leave. It _might be_ for the beginning assuming that validators won't change a lot so we won't expect this map to grow, but on the long run I don't feel comfortable leaving this unbounded.